### ISSUE TITLE
Lost type annotations of global import

### ISSUE THREAD

**Issue by fgarcia** [facebook/flow#2325]
I am losing the type information for a file I own. Let's say I have this file:

```
// @flow (file: foo.js)
const F = {
  foo: 1,
  bar: 2,
}

export default F;
```

Now I use that file somewhere else:

```
// @flow (file: demo.js)
import F from './foo'

F.something    // this will raise an error
```

Since I use that file so much, I promote it into a global variable and declare that in my flow config

```
// (file: globals.js.flow)
import MyF from 'foo.js'
declare var F : typeof MyF;
```

Also I remove all other imports of `F` in my code:
```
// @flow (file: demo.js)
F.something    // this is no longer an error ????
```

Now all type checks are lost. I've tried importing/exporting/declaring the type of the global var `F` in several ways, but always all type checks are gone. 

Which is the right syntax to preserve the type annotations?

**Comment by vkurchatkin** [comment]
As far as I remember, imports don't work in lib files

**Comment by fgarcia** [comment]
I am hopping for a way to define a global variable without re-typing all its type annotations. 

The import line above was just one example of my failed attemts

**Comment by ryyppy** [comment]
Wait, shadow files (flow.js) do not bleed through the global scope, right? You are talking about a libdef file, which is involved via [LIB]?

If so, import doesn't work in libdef files (yet)... I am afraid you need to duplicate type information for now

**Comment by samwgoldman** [comment]
Yeah, I think there's lots of potential for confusion between .js.flow files and libdef files right now. Based on the issue, I suspect that's what's going on here. Going to tag this as needs docs, but hopefully @ryyppy and @vkurchatkin can help unblock you :)

**Comment by fgarcia** [comment]
Oh, so I am kind of heartbroken here :(

According the docs [here](https://flowtype.org/docs/third-party.html) this is the way to declare a global variable:

```
// interfaces/underscore.js
declare class Underscore {
  findWhere<T>(list: Array<T>, properties: {}): T;
}

declare var _: Underscore;

```

That is the task I am having problems to replicate with my own module.

Correct me if I am wrong, but if I understood correctly, it is not possible to define a typed global variable unless I use a declaration file/syntax. Flow will not recognize the types from an already annotated module (in code) for a global variable. Is that right?

**Comment by ryyppy** [comment]
> it is not possible to define a typed global variable unlses I use a declaration file / syntax

To define a globally typed variable, you need a libdef file.

Example `.flowconfig`:

```
[ignore]

[include]

[libs]
flow-typed/myF.js

```

Then you put this in your `flow-typed/myF.js`:

```javascript
declare type F = {
   foo: number,
   bar: number,
};
```


Now, if you have some file like `foo.js`:

```javascript
const f: F = {
  foo: 'nope',
};
```

It will realize that `foo` is not a number, so it recognizes the global type `F`.

For the `flow.js`, if you want to vendor your type declarations from the ORIGINAL source file, just copy the source file and put the `flow.js` ending in it... make sure that e.g. `src/myF.js` is then copied to `src/myF.flow.js`.

If you do that `flow.js` generation, you have following situation:

```javascript
import F from './src/myF';

// Flow will now the type of F, since it picked up the flow.js stuff
(F.foo: number);

```

The latter example is not very useful if you are not stripping the flowtypes from the original `src/myF.js` of course, otherwise you would just import it normally and flow will pick up the types.

Does this answer your questions?

**Comment by ryyppy** [comment]
To clarify, if you create a file `globals.js.flow`, you will have to actually import this file, to make flow realize that.

You need to create a libdef file, put it into `flow-typed/globals.js` and add the `declare type F = {...}` stuff, to make it globally available

**Comment by fgarcia** [comment]
@ryyppy Your solution did work!

Although I am going to give up on using globals with Flow altogether. 

After all the work setting up Flow for this case, I've just noticed that all imports within `./src/myF` *fail silently* again, so I did not realize all imported types where turned into `any` again.

Now that I am importing the global I defined that way, lots of undetected type errors are showing up.
